sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/options/FileParallelUploadOptions.java

```diff
@@ -41,6 +43,21 @@ public FileParallelUploadOptions(Flux<ByteBuffer> dataFlux) {
         this.dataFlux = dataFlux;
         this.dataStream = null;
         this.length = null;
+        this.data = null;
+    }
+
+    /**
+     * Constructs a new {@code FileParallelUploadOptions}.
+     *
+     * @param data The {@link BinaryData} to write to the file.
+     */
+    public FileParallelUploadOptions(BinaryData data) {
+        StorageImplUtils.assertNotNull("data must not be null", data);
+        StorageImplUtils.assertNotNull("data must have defined length", data.getLength());
```

Comment:
Why do we have this check? The other overload that takes Flux<ByteBuffer> as input doesn't require length.

Reply:
Good catch, I'll remove this check since not needed.